.github/workflows/publish-on-release.yml

```diff
@@ -67,10 +70,11 @@ jobs:
 
   publish-website:
     name: Publish to the production website
-    needs: [publish-assets]
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
+        with:
+          ref: release
```

Comment:
@samhotep As far as I understand it would checkout `release` branch, and push back to it.

What we need is checkout of `main` branch, and force pushing its contents to `release` branch.

Reply:
I think the issue we have comes from the fact that checkout action only fetches a single commit by default (https://github.com/actions/checkout) so it doesn't have a branch to push.

There are probably 2 ways of working around that:

Fetching whole history with all branches: https://github.com/actions/checkout#Fetch-all-history-for-all-tags-and-branches

Or specifying that we want to force push the commit we are on `git push --force upstream HEAD:release`

Reply:
I see..
So I've been playing around with both options, and the [first one](https://github.com/samhotep/codespaces-actions-playground/actions/runs/6171992084/workflow) seems much easier and cleaner, so maybe we can go with that? Also I couldn't get the [second one](https://github.com/samhotep/codespaces-actions-playground/actions/runs/6171979230/workflow) to work without manually authenticating git :(

Reply:
Thanks for checking that out. It still wont solve the protected branch issue I think.